src/storage/engine.go

```diff
@@ -92,3 +95,27 @@ func (mse *MapStorageEngine) Delete(keys []string) (int, error) {
 
 	return deletedCount, nil
 }
+
+func (mse *MapStorageEngine) AtomicDelta(key string, delta int64) (int64, error) {
+	mse.mu.Lock()
+	defer mse.mu.Unlock()
+
+	valCtr, ok := mse.store[key]
+	if !ok {
+		// counter doesn't exist yet, forcefully set it to the delta value and return the same
```

Comment:
This is a near way to implement default zero. What if the delta is negative?

Reply:
The delta can be negative, for example, in the `DECR` command, the logic, if the key is not present, is to:
1. set to 0
2. decrement by 1 (delta will be sent as -1)
3. return -1

ref: https://redis.io/docs/latest/commands/decr/

This approach skips the set to 0 and directly sets to the expected value.

In future, the application may also support https://redis.io/docs/latest/commands/decrby/ and https://redis.io/docs/latest/commands/incrby/ which may use non-unit deltas in which case, the behavior would be the same.